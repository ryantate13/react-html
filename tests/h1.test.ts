import {h1} from '../src';

describe('h1', () => {
    it('renders with children', () => {
        expect(h1({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(h1({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(h1({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(h1()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(h1({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(h1({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(h1({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(h1({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(h1({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(h1({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(h1({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(h1({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(h1({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(h1({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(h1({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(h1({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(h1({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(h1({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(h1({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(h1({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(h1({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(h1({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(h1({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(h1({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(h1({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(h1({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(h1({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(h1({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(h1({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(h1({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(h1({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(h1({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(h1({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(h1({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(h1({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(h1({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(h1({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(h1({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(h1({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(h1({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(h1({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(h1({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(h1({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(h1({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(h1({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(h1({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(h1({translate: "no"})).toMatchSnapshot();
    });
});