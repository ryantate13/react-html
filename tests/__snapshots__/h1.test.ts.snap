// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1 renders arrays of children 1`] = `
<h1>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</h1>
`;

exports[`h1 renders with accessKey="a,b,c,d" 1`] = `
<h1
  accessKey="a,b,c,d"
/>
`;

exports[`h1 renders with autoCapitalize="characters" 1`] = `
<h1
  autoCapitalize="characters"
/>
`;

exports[`h1 renders with autoCapitalize="none" 1`] = `
<h1
  autoCapitalize="none"
/>
`;

exports[`h1 renders with autoCapitalize="off" 1`] = `
<h1
  autoCapitalize="off"
/>
`;

exports[`h1 renders with autoCapitalize="on" 1`] = `
<h1
  autoCapitalize="on"
/>
`;

exports[`h1 renders with autoCapitalize="sentences" 1`] = `
<h1
  autoCapitalize="sentences"
/>
`;

exports[`h1 renders with autoCapitalize="words" 1`] = `
<h1
  autoCapitalize="words"
/>
`;

exports[`h1 renders with children 1`] = `
<h1>
  TEST
</h1>
`;

exports[`h1 renders with className="foo bar baz" 1`] = `
<h1
  className="foo bar baz"
/>
`;

exports[`h1 renders with contentEditable="false" 1`] = `
<h1
  contentEditable="false"
/>
`;

exports[`h1 renders with contentEditable="true" 1`] = `
<h1
  contentEditable="true"
/>
`;

exports[`h1 renders with dir="auto" 1`] = `
<h1
  dir="auto"
/>
`;

exports[`h1 renders with dir="ltr" 1`] = `
<h1
  dir="ltr"
/>
`;

exports[`h1 renders with dir="rtl" 1`] = `
<h1
  dir="rtl"
/>
`;

exports[`h1 renders with draggable="false" 1`] = `
<h1
  draggable="false"
/>
`;

exports[`h1 renders with draggable="true" 1`] = `
<h1
  draggable="true"
/>
`;

exports[`h1 renders with empty attributes 1`] = `<h1 />`;

exports[`h1 renders with hidden=false 1`] = `
<h1
  hidden={false}
/>
`;

exports[`h1 renders with hidden=true 1`] = `
<h1
  hidden={true}
/>
`;

exports[`h1 renders with id="TEST" 1`] = `
<h1
  id="TEST"
/>
`;

exports[`h1 renders with inputMode="decimal" 1`] = `
<h1
  inputMode="decimal"
/>
`;

exports[`h1 renders with inputMode="email" 1`] = `
<h1
  inputMode="email"
/>
`;

exports[`h1 renders with inputMode="none" 1`] = `
<h1
  inputMode="none"
/>
`;

exports[`h1 renders with inputMode="numeric" 1`] = `
<h1
  inputMode="numeric"
/>
`;

exports[`h1 renders with inputMode="search" 1`] = `
<h1
  inputMode="search"
/>
`;

exports[`h1 renders with inputMode="tel" 1`] = `
<h1
  inputMode="tel"
/>
`;

exports[`h1 renders with inputMode="text" 1`] = `
<h1
  inputMode="text"
/>
`;

exports[`h1 renders with inputMode="url" 1`] = `
<h1
  inputMode="url"
/>
`;

exports[`h1 renders with is="foo" 1`] = `
<h1
  is="foo"
/>
`;

exports[`h1 renders with itemID="https://google.com/" 1`] = `
<h1
  itemID="https://google.com/"
/>
`;

exports[`h1 renders with itemProp="https://google.com/" 1`] = `
<h1
  itemProp="https://google.com/"
/>
`;

exports[`h1 renders with itemRef="foo bar" 1`] = `
<h1
  itemRef="foo bar"
/>
`;

exports[`h1 renders with itemScope=false 1`] = `
<h1
  itemScope={false}
/>
`;

exports[`h1 renders with itemScope=true 1`] = `
<h1
  itemScope={true}
/>
`;

exports[`h1 renders with itemType="https://google.com/" 1`] = `
<h1
  itemType="https://google.com/"
/>
`;

exports[`h1 renders with lang="" 1`] = `
<h1
  lang=""
/>
`;

exports[`h1 renders with lang="es-419" 1`] = `
<h1
  lang="es-419"
/>
`;

exports[`h1 renders with slot="TEST" 1`] = `
<h1
  slot="TEST"
/>
`;

exports[`h1 renders with spellCheck="false" 1`] = `
<h1
  spellCheck="false"
/>
`;

exports[`h1 renders with spellCheck="true" 1`] = `
<h1
  spellCheck="true"
/>
`;

exports[`h1 renders with style={"color":"#000000"} 1`] = `
<h1
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`h1 renders with tabIndex=1 1`] = `
<h1
  tabIndex={1}
/>
`;

exports[`h1 renders with title="TEST" 1`] = `
<h1
  title="TEST"
/>
`;

exports[`h1 renders with translate="no" 1`] = `
<h1
  translate="no"
/>
`;

exports[`h1 renders with translate="yes" 1`] = `
<h1
  translate="yes"
/>
`;

exports[`h1 renders without attributes 1`] = `<h1 />`;
